src/braket/default_simulator/simulator.py

```diff
@@ -57,7 +59,32 @@
 )
 
 
-class BaseLocalSimulator(BraketSimulator):
+class OpenQASMSimulator(BraketSimulator, ABC):
```

Comment:
Hm I feel like we had this in the initial implementation and got rid of this. Since we're trending towards openqasm, what if we just added the method to the base simulator class with a default NotImplementedError implementation?

Reply:
Thing is, `BaseLocalSimulator` is already mostly implemented; it parses OpenQASM into a default sim `Circuit` object, and the subclasses are meant to be internal to this repo. `OpenQASMSimulator` is a public interface (I'll add more documentation) for developers to implemented if they want to use the BDK with their own backend.

Reply:
Hm, should this PR update `BaseLocalSimulator` to inherit from `OpenQASMSimulator`?

Reply:
It already does!
```python
class BaseLocalSimulator(OpenQASMSimulator):
```

Reply:
Oh got it, so basically this PR is just changing
`BraketSimulator`->`BaseLocalSimulator`
to
`BraketSimulator`->`OpenQASMSimulator`->`BaseLocalSimulator`
and adding some general methods to the middle for other backends to customize

Reply:
Exactly! I've added more detailed instructions in the docstring.

Reply:
does adding ABC here imply that someone who implemented a subclass of `BaseLocalSimulator` would break when updating the local sim? Would we prefer to use NotImplementedErrors?

Reply:
`BaseLocalSimulator` already inherited directly from `BraketSimulator` which is an ABC, before their change, so this won't change anything